services/src/main/java/org/keycloak/email/freemarker/FreeMarkerEmailTemplateProvider.java

```diff
@@ -217,13 +219,19 @@ protected EmailTemplate processTemplate(String subjectKey, List<Object> subjectA
                 textBody = freeMarker.processTemplate(attributes, textTemplate, theme);
             } catch (final FreeMarkerException e) {
                 textBody = null;
+                logger.warn("Failed to template plain text email.", e);
```

Comment:
I'd say this should just do:
`throw new EmailException("Failed to template plain text email.", e);`

Reply:
I don't think so, since the email can be sent in html.

Reply:
Keycloak expects to send both text/html emails, so if it can't process both templates that is still an error and should be handled as such. Not throwing the exception can lead to situations where people believe the templates are working, but they are not.

Reply:
We do plan to add an option to allow sending text, html or both, but since both is expected now not being able to process both templates should be handle as an error.